tikv-client/src/main/java/com/pingcap/tikv/txn/LockResolverClient.java

```diff
@@ -0,0 +1,288 @@
+/*
+ *
+ * Copyright 2017 PingCAP, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.pingcap.tikv.txn;
+
+import com.google.protobuf.ByteString;
+import com.pingcap.tikv.AbstractGRPCClient;
+import com.pingcap.tikv.TiSession;
+import com.pingcap.tikv.exception.KeyException;
+import com.pingcap.tikv.exception.RegionException;
+import com.pingcap.tikv.kvproto.Kvrpcpb.CleanupRequest;
+import com.pingcap.tikv.kvproto.Kvrpcpb.CleanupResponse;
+import com.pingcap.tikv.kvproto.Kvrpcpb.ResolveLockRequest;
+import com.pingcap.tikv.kvproto.Kvrpcpb.ResolveLockResponse;
+import com.pingcap.tikv.kvproto.Metapb.Store;
+import com.pingcap.tikv.kvproto.TikvGrpc;
+import com.pingcap.tikv.kvproto.TikvGrpc.TikvStub;
+import com.pingcap.tikv.kvproto.TikvGrpc.TikvBlockingStub;
+import com.pingcap.tikv.operation.KVErrorHandler;
+import com.pingcap.tikv.region.RegionErrorReceiver;
+import com.pingcap.tikv.region.TiRegion;
+import com.pingcap.tikv.region.TiRegion.RegionVerID;
+import com.pingcap.tikv.util.BackOffer;
+import com.pingcap.tikv.util.TsoUtils;
+import io.grpc.ManagedChannel;
+import org.apache.log4j.Logger;
+
+import java.util.*;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import java.util.function.Supplier;
+
+import static com.pingcap.tikv.util.BackOffFunction.BackOffFuncType.BoRegionMiss;
+
+// LockResolver resolves locks and also caches resolved txn status.
+public class LockResolverClient extends AbstractGRPCClient<TikvBlockingStub, TikvStub> implements RegionErrorReceiver {
+  // ResolvedCacheSize is max number of cached txn status.
+  private static final long resolvedTxnCacheSize = 2048;
```

Comment:
For those variable constants, I recommend you to use snake case rather than camel case.
```suggestion
  private static final long RESOLVED_TXN_CACHE_SIZE = 2048;
```